js/opening-hours.functions.js

```diff
@@ -124,103 +443,197 @@ OpeningHours.prototype.init = function () {
 OpeningHours.prototype.formattedDate = function (dateString) {
   if (dateString === 'today') {
     var today = new Date();
-    dateString = today.toISOString().slice(0, 10);
+    return this.formatDate(today);
   }
+
   if (dateString === 'tomorrow') {
     var tomorrow = new Date();
     tomorrow.setDate(tomorrow.getDate() + 1);
-    dateString = tomorrow.toISOString().slice(0, 10);
+    return this.formatDate(tomorrow);
   }
 
   var date = !dateString ? new Date() : new Date(dateString);
 
   try {
-    var iso = date.toISOString();
-  } catch (err) {
+    return this.formatDate(date);
+  }
+  catch (err) {
     date = new Date();
-    iso = date.toISOString();
+    return this.formatDate(date);
   }
+};
 
-  return iso.slice(0, 10);
+/**
+ * Render the widget for the given item.
+ *
+ * @param {OpeningHoursItem} item
+ *   The item to render.
+ */
+OpeningHours.prototype.renderItemWidget = function (item) {
+  if (!item.isWidgetable()) {
+    return false;
+  }
+
+  this.getTitle(item);
+  this.constructRequest(
+    this.constructWidget(item, this),
+    item
+  );
 };
 
 /**
  * Create the proper API request URI.
  *
+ * @param {function} callback
+ *   The callback when the request is processed.
+ * @param {OpeningHoursItem} item
+ *   The opening hours item to render the widget for.
+ *
  * @returns {string}
  *   The request URI.
  */
-OpeningHours.prototype.constructRequest = function (cb) {
-  var requestOptions = {
-    parameters: {
-      date: this._current.dataset.date,
-      language: this.settings.language
-    }
-  };
-
-  switch (this._current.dataset.type) {
+OpeningHours.prototype.constructRequest = function (callback, item) {
```

Comment:
Would make more sense to switch these parameters around, since the item is required and the callback is optional.

```suggestion
 OpeningHours.prototype.constructRequest = function (item, callback) {
```

Reply:
Changed.